pkg/operator/sync.go

```diff
@@ -126,6 +138,59 @@ func (optr *Operator) syncTerminationHandler(config *OperatorConfig) error {
 	return nil
 }
 
+func (optr *Operator) syncWebhookConfiguration(config *OperatorConfig) error {
+	err := optr.syncValidatingWebhook(optr.kubeClient.AdmissionregistrationV1().ValidatingWebhookConfigurations(), newValidatingWebhook(config))
+	if err != nil {
+		return err
+	}
+	err = optr.syncMutatingWebhook(optr.kubeClient.AdmissionregistrationV1().MutatingWebhookConfigurations(), newMutatingWebhook(config))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (optr *Operator) syncValidatingWebhook(client admissionclientv1.ValidatingWebhookConfigurationInterface, hook *admissionregistrationv1.ValidatingWebhookConfiguration) error {
+	validHook, err := optr.validatingWebhookLister.Get(hook.Name)
+	if apierrors.IsNotFound(err) {
+		_, err = client.Create(context.Background(), hook, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+		return nil
+	} else if err != nil {
+		return err
+	}
+	if !equality.Semantic.DeepDerivative(hook.Webhooks, validHook.Webhooks) {
+		_, err = client.Update(context.Background(), hook, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
I've not seen `DeepDerivative` used for this before, have you verified that this doesn't hot loop updating?

Reply:
It has it's drawbacks, so it won't be able to detect a change if there is a deleted field in the new webhook configuration, but the field is present on existing resource.